spec/requests/embedded_shopfronts_headers_spec.rb

```diff
@@ -61,5 +61,20 @@
         expect(response.headers['Content-Security-Policy']).to eq "frame-ancestors 'none'"
       end
     end
+
+    context "with www prefix" do
+      before do
+        Spree::Config[:embedded_shopfronts_whitelist] = "example.com external-site.com"
+        allow_any_instance_of(ActionDispatch::Request).to receive(:referer).and_return('http://www.external-site.com/shop?embedded_shopfront=true')
```

Comment:
you have access to `request` here so there's no need to use neither `allow_any_instance` nor refer to `ActionDispatch::Request`.